spec/services/hyrax/collections/migration_service_spec.rb

```diff
@@ -36,24 +44,107 @@
+      end
+
       it "doesn't change the collection" do
         expect(collection.collection_type_gid).to eq collection_type_gid
         expect(Hyrax::PermissionTemplate.find_by!(source_id: collection.id).id).to eq permission_template.id
-        expect_access(permission_template.id, 'user', :manage, edit_users)
+        expect_access(permission_template.id, 'user', :manage, [user.user_key, manager1.user_key])
 
         Hyrax::Collections::MigrationService.migrate_all_collections
 
         expect(collection.collection_type_gid).to eq collection_type_gid
         expect(Hyrax::PermissionTemplate.find_by!(source_id: collection.id).id).to eq permission_template.id
-        expect_access(permission_template.id, 'user', :manage, edit_users)
+        expect_access(permission_template.id, 'user', :manage, [user.user_key, manager1.user_key])
+      end
+    end
+
+    context 'when legacy adminsets are found (e.g. adminsets created before Hyrax 2.1.0)' do
+      let!(:as_none) { build(:no_solr_grants_adminset, id: 'as_none', user: user) }
+      let!(:as_vu) { build(:no_solr_grants_adminset, id: 'as_vu', user: user, with_permission_template: { view_users: [reader1.user_key, reader2.user_key] }) }
+      let!(:as_vg) { build(:no_solr_grants_adminset, id: 'as_vg', user: user, with_permission_template: { view_groups: ['read_group_1', 'read_group_2'] }) }
+      let!(:as_du) { build(:no_solr_grants_adminset, id: 'as_du', user: user, with_permission_template: { deposit_users: [depositor1.user_key, depositor2.user_key] }) }
+      let!(:as_dg) { build(:no_solr_grants_adminset, id: 'as_dg', user: user, with_permission_template: { deposit_groups: ['deposit_group_1', 'deposit_group_2'] }) }
+      let!(:as_mu) { build(:no_solr_grants_adminset, id: 'as_mu', user: user, with_permission_template: { manage_users: [editor1.user_key, editor2.user_key] }) }
+      let!(:as_mg) { build(:no_solr_grants_adminset, id: 'as_mg', user: user, with_permission_template: { manage_groups: ['edit_group_1', 'edit_group_2'] }) }
+
+      before do
+        allow(AdminSet).to receive(:all).and_return([as_none, as_vu, as_vg, as_du, as_dg, as_mu, as_mg])
+      end
+
+      it 'sets read and edit access in solr doc' do # rubocop:disable RSpec/ExampleLength
+        AdminSet.all.each do |adminset|
+          expect(adminset.edit_users).to include(user.user_key)
+          expect(adminset.read_users).to eq []
+        end
+
+        Hyrax::Collections::MigrationService.migrate_all_collections
+
+        AdminSet.all.each do |adminset|
+          pt_id = Hyrax::PermissionTemplate.find_by!(source_id: adminset.id)
+          expect(pt_id).not_to be_nil
+          expect_solr_access(pt_id, 'user', :manage, adminset.edit_users)
+          expect_solr_access(pt_id, 'group', :manage, adminset.edit_groups)
+          expect_solr_access(pt_id, 'user', :view, adminset.read_users)
+          expect_solr_access(pt_id, 'group', :view, adminset.read_groups)
```

Comment:
Are lines 99-100 supposed to be for depositor user and group?  Currently they are a repeat of lines 101-102.  It looks like this the case in the other new tests below as well for both pre and post migration tests.

Reply:
Good catch.

Reply:
It looks like these still need to be changed to :deposit.